builtin/providers/vsphere/resource_vsphere_virtual_machine.go

```diff
@@ -0,0 +1,1062 @@
+			if v, ok := d.GetOk(prefix + ".datastore"); ok {
+				vm.datastore = v.(string)
+			}
+		} else {
+			if v, ok := d.GetOk(prefix + ".size"); ok {
+				disks[i].size = int64(v.(int))
+			} else {
+				return fmt.Errorf("Size argument is required.")
+			}
+		}
+		if v, ok := d.GetOk(prefix + ".iops"); ok {
+			disks[i].iops = int64(v.(int))
+		}
+	}
+	vm.hardDisks = disks
+	log.Printf("[DEBUG] disk init: %v", disks)
+
+	if vm.template != "" {
+		err := vm.deployVirtualMachine(client)
+		if err != nil {
+			return fmt.Errorf("error: %s", err)
+		}
+	} else {
+		err := vm.createVirtualMachine(client)
+		if err != nil {
+			return fmt.Errorf("error: %s", err)
+		}
+	}
+
+	if _, ok := d.GetOk("network_interface.0.ip_address"); !ok {
+		if v, ok := d.GetOk("boot_delay"); ok {
+			stateConf := &resource.StateChangeConf{
+				Pending:    []string{"pending"},
+				Target:     "active",
+				Refresh:    waitForNetworkingActive(client, vm.datacenter, vm.name),
+				Timeout:    600 * time.Second,
+				Delay:      time.Duration(v.(int)) * time.Second,
+				MinTimeout: 2 * time.Second,
+			}
+
+			_, err := stateConf.WaitForState()
+			if err != nil {
+				return fmt.Errorf("error: %s", err)
+			}
+		}
+	}
+	d.SetId(vm.name)
+	log.Printf("[INFO] Created virtual machine: %s", d.Id())
+
+	return resourceVSphereVirtualMachineRead(d, meta)
+}
+
+func resourceVSphereVirtualMachineRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*govmomi.Client)
+	dc, err := getDatacenter(client, d.Get("datacenter").(string))
+	if err != nil {
+		return err
+	}
+	finder := find.NewFinder(client.Client, true)
+	finder = finder.SetDatacenter(dc)
+
+	vm, err := finder.VirtualMachine(context.TODO(), d.Get("name").(string))
+	if err != nil {
+		log.Printf("[ERROR] Virtual machine not found: %s", d.Get("name").(string))
+		d.SetId("")
+		return nil
+	}
+
+	var mvm mo.VirtualMachine
+
+	collector := property.DefaultCollector(client.Client)
+	if err := collector.RetrieveOne(context.TODO(), vm.Reference(), []string{"guest", "summary", "datastore"}, &mvm); err != nil {
+		log.Printf("[ERROR] %#v", err)
+	}
+
+	log.Printf("[DEBUG] %#v", dc)
+	log.Printf("[DEBUG] %#v", mvm.Summary.Config)
+	log.Printf("[DEBUG] %#v", mvm.Guest.Net)
+
+	networkInterfaces := make([]map[string]interface{}, 0)
+	for _, v := range mvm.Guest.Net {
+		if v.DeviceConfigId >= 0 {
+			log.Printf("[DEBUG] %#v", v.Network)
+			networkInterface := make(map[string]interface{})
+			networkInterface["label"] = v.Network
+			if len(v.IpAddress) > 0 {
+				log.Printf("[DEBUG] %#v", v.IpAddress[0])
+				networkInterface["ip_address"] = v.IpAddress[0]
+
+				m := net.CIDRMask(v.IpConfig.IpAddress[0].PrefixLength, 32)
+				subnetMask := net.IPv4(m[0], m[1], m[2], m[3])
+				networkInterface["subnet_mask"] = subnetMask.String()
+				log.Printf("[DEBUG] %#v", subnetMask.String())
+			}
+			networkInterfaces = append(networkInterfaces, networkInterface)
+		}
+	}
+	d.Set("network_interface", networkInterfaces)
```

Comment:
`d.Set` returns an error value that we ignore on primitive type, but we check it on complex types like this for an extra layer of safety.